File: py/1/ex1/main.py
import re

with open("./input.txt") as file:
    data = file.read().split("\n")


def first_last_digit(line: str) -> int:
    digits = re.findall("[0-9]", line)
    return int(digits[0] + digits[-1])


print(sum([first_last_digit(line) for line in data]))

File: ts/1/ex1/main.ts
import { readFile } from "fs/promises"

function sumAll(array: number[]): number {
    return array.reduce((prev, curr) => prev + curr, 0)
}

function firstLastDigit(line: string): number | null {
    const digits = line.match(new RegExp("[0-9]", "g"))

    if (digits === null) return null

    return Number(digits[0] + digits[digits.length - 1])
}

readFile("./input.txt")
    .then(data => {
        console.log(
            sumAll(data.toString().split("\n").map(line => {
                return firstLastDigit(line) ?? 0
            }))
        )
    })